akka-http/src/main/scala/akka/http/javadsl/server/directives/SecurityDirectives.scala

```diff
@@ -259,6 +259,10 @@ abstract class SecurityDirectives extends SchemeDirectives {
   /**
    * Creates a `Basic` [[HttpChallenge]] for the given realm.
    */
-  def challengeFor(realm: String): HttpChallenge = HttpChallenge.create("Basic", realm)
+  def basicChallengeFor(realm: String): HttpChallenge = HttpChallenge.create("Basic", realm)
```

Comment:
I like your idea! :+1:  And yes, by applying your thought, I think we can remove these methods safely.

To make it sure, is the code below what you meant? (FYI, I do't have any preference on method names. Plz point out if you have sth in mind :) ) If the change is fine, I'm supposed to `rebase -i` and fix my first commits. 
https://github.com/monkey-mas/akka/commit/a0b4bf14771ea4c91cea86ad979898ec9ff57dec